presto-main/src/main/java/com/facebook/presto/operator/window/WindowAnnotationsParser.java

```diff
@@ -62,6 +62,8 @@ private static SqlWindowFunction parse(Class<? extends WindowFunction> clazz, Wi
                 argumentTypes,
                 false);
 
-        return new SqlWindowFunction(new ReflectionWindowFunctionSupplier<>(signature, clazz));
+        boolean canIgnoreNulls = window.name().equalsIgnoreCase("lag") || window.name().equalsIgnoreCase("lead");
```

Comment:
Instead of hard-coding it for these two, we should discover it from function definitions so that it can apply to any function that declares they support this option (e.g., via an annotation)